src/net_processing.cpp

```diff
@@ -1252,6 +1253,47 @@ bool PeerManagerImpl::BlockRequestAllowed(const CBlockIndex* pindex)
            (GetBlockProofEquivalentTime(*pindexBestHeader, *pindex, *pindexBestHeader, m_chainparams.GetConsensus()) < STALE_RELAY_AGE_LIMIT);
 }
 
+bool PeerManagerImpl::FetchBlock(NodeId id, const CBlockIndex& index)
+{
+    PeerRef peer = GetPeerRef(id);
+    if (peer == nullptr) return false;
+    if (fImporting && !fReindex) return false;
```

Comment:
Why not match the logic in `ProcessMessage()` when receiving a block:

```suggestion
    if (fImporting || fReindex) return false;
```

Reply:
I think I copied this one from another method. But fewer negations make sense.